commands/create-account.js

```diff
@@ -34,6 +36,34 @@ module.exports = {
 async function createAccount(options) {
     await eventtracking.track(eventtracking.EVENT_ID_CREATE_ACCOUNT_START, { nodeUrl: options.nodeUrl });
     // NOTE: initialBalance is passed as part of config here, parsed in middleware/initial-balance
+    // periods are disallowed in top-level accounts and can only be used for subaccounts
+    const splitAccount = options.accountId.split('.');
+
+    const splitMaster = options.masterAccount.split('.');
+    const masterRootTLA = splitMaster[splitMaster.length - 1];
+    if (splitAccount.length === 1) {
+        // TLA (bob-with-at-least-maximum-characters)
+        if (splitAccount[0].length < TLA_MIN_LENGTH) {
+            console.log(`Top-level accounts must be greater than ${TLA_MIN_LENGTH} characters.\n` +
+              'Note: this is for advanced usage only. Typical account names are of the form:\n' +
+              'app.alice.test, where the masterAccount shares the top-level account (.test).'
+            );
+            return;
+        }
+    } else if (splitAccount.length > 1) {
+        // Subaccounts (short.alice.near, even.more.bob.test, and eventually peter.potato)
+        // Check that master account TLA matches
+        const accountTLA = splitAccount.filter((n, i) => i !== 0).join('.');
+        if (accountTLA !== options.masterAccount) {
+            console.log(`New account doesn't share the same top-level account. Expecting account name to end in ".${options.masterAccount}"`);
+            return;
+        }
+
+        // Warn user if account seems to be using wrong network, where TLA is captured in config
+        if (Object.prototype.hasOwnProperty.call(options, 'tla') && masterRootTLA !== options.tla) {
```

Comment:
@mikedotexe is there any reason why we need to treat `tla` differently than `masterAccount`? Like it's not that all accounts on mainnet will have to end with `.near`.

It's just when you want to create `account_name.near` you need to have access to `near` as master account (either directly or through contract helper).

Reply:
So the purpose of this check is to send a console log (not error) when people try to use `masterAccount` from a different network.
In an ideal world (and I included this in the shell NEP) we would have `{"account_id":…, "networkId":…}` so we can check if they're trying to use the wrong network.
There have been a lot of people (especially validators/stakers) that have forgotten to use the `NODE_ENV=betanet` (or `NEAR_ENV=betanet`) so the only purpose of this is to make sure they're on the right network.
I believe what you're talking about is a check that is right above this piece of code. Right before this we check if the new account has the same TLA as the masterAccount like you're suggesting. That check will `return` unlike this one which is just a "NOTE"

Reply:
> In an ideal world (and I included this in the shell NEP) we would have {"account_id":…, "networkId":…} so we can check if they're trying to use the wrong network.

let's create issue for this and link in `TODO` from here.